src/Google.Api.Gax/ServiceSettingsBase.cs

```diff
@@ -44,6 +58,7 @@ public abstract class ServiceSettingsBase
         {
             settings.CallSettings = CallSettings.Clone();
             settings.Clock = Clock;
+            settings.UserAgent = UserAgent;
```

Comment:
Ah, hadn't spotted that - yes, of course. Bit of a pain that it will have to regenerate it on each clone though. Can we avoid that? How often do we clone service settings?